File: ver-android/app/src/main/kotlin/org/opengamestudio/VM.kt
package org.opengamestudio

import android.content.Context
import androidx.compose.runtime.*

object VM {
    var androidContext: Context? = null

    val isBudgetTabSelected = mutableStateOf(false)
    val isSettingsTabSelected = mutableStateOf(false)

    val logs = mutableStateListOf<String>()

    val morningBalance = mutableStateOf("")
    val morningBalancePasteTitle = mutableStateOf("Paste")
    val morningBalancePlaceholder = mutableStateOf("Morning balance")

    val result = mutableStateOf("TODO-Result")
    val resultCopyTitle = mutableStateOf("Copy")

    val saveTitle = mutableStateOf("Save")

    val spent = mutableStateOf("")
    val spentPasteTitle = mutableStateOf("Paste")
    val spentPlaceholder = mutableStateOf("Spent")
}

File: ver-ios/src/VM.swift
import Combine

class VM: ObservableObject {
    @Published var morningBalance = ""
    @Published var morningBalancePasteTitle = "Paste"
    @Published var morningBalancePlaceholder = "Morning balance"

    @Published var spent = ""
    @Published var spentPasteTitle = "Paste"
    @Published var spentPlaceholder = "Spent"

    @Published var result = "TODO-Result"
    @Published var resultCopyTitle = "Copy"

    static private(set) weak var singleton: VM?

    init() {
        Self.singleton = self
    }
}
